lib/spack/docs/packaging_guide.rst

```diff
@@ -2315,17 +2315,27 @@ relationships between packages.
 Version ranges
 ^^^^^^^^^^^^^^
 
-Although some packages require a specific version for their dependencies,
-most can be built with a range of versions. For example, if you are
-writing a package for a legacy Python module that only works with Python
-2.4 through 2.6, this would look like:
+Although some packages require a specific version for their dependencies, most
+can be built with a range of versions. The simplest way to construct a range is
+to use a partial version number. If all of the packages are versioned with three
+numbers, ``x.y.z``, then ``x.y`` can match any version beginning with ``x.y.``
+(the third version number is a free variable). This is different from previous
+versions of Spack where ``x.y`` only match exactly ``x.y``, which might not even
+exist.
+
+One can specify looser ranges by writing two versions separated by a colon. The
+left-hand version is the first acceptable version, while the right-hand version
+is the last acceptable version. If either version is partially specified, then the
+unspecified digits are interpreted to make the range as loose as possible. For
+example, if you are writing a package for a legacy Python module that only works
+with Python 2.4.x through 2.6.x, this would look like:
 
 .. code-block:: python
 
    depends_on('python@2.4:2.6')
 
 Version ranges in Spack are *inclusive*, so ``2.4:2.6`` means any version
-greater than or equal to ``2.4`` and up to and including any ``2.6.x``. If
+greater than or equal to ``2.4.0`` and lesser than or equal to ``2.6.x``. If
```

Comment:
In Spack, alphabetic characters sort as older than numeric characters (with the exception of [infinity versions](https://github.com/spack/spack/blob/v0.19.0/lib/spack/spack/version.py#L64)). So technically, `2.4:2.6` means `2.4.a:2.6.develop`, where `2.4.a < 2.4.0`. Not sure how technical we want to get here. I would prefer "greater than or equal to `2.4`" over `2.4.0` since `2.4.0` is technically incorrect.

Reply:
Good point.